src/cache/lru-cache.ts

```diff
@@ -97,3 +97,22 @@ export class LRUCache implements Map<string, string> {
     return this;
   }
 }
+
+/**
+ * Time-aware, least-recently-used (TLRU), variant of LRU where entries have valid lifetime.
+ * @param {number} maxSize - Maximum cache size
+ * @param {number} ttl - Time in milliseconds after which cache entry will evict itself
+ **/
+export class TLRUCache extends LRUCache {
+  constructor(readonly maxSize: number, readonly ttl: number) {
+    super(maxSize);
+  }
+
+  set(key: string, value: string): this {
+    const cache = super.set(key, value);
+    setTimeout(() => {
+      this.delete(key);
+    }, this.ttl);
```

Comment:
Yeah, you are right. I've created secondary registry that keeps track of timeouts by cache key, so that we can 'refresh' the ttl when entry is being overwritten.